### PR TITLE
Fix mdx file path parsing of dots

### PR DESCRIPTION
**Pull request by kepae**
Addresses #2640.

### CODE REVIEW

File: site/gatsby-site/plugins/gatsby-theme-i18n/gatsby-node.js [hunk 1]
```diff
@@ -220,27 +220,45 @@ exports.onCreatePage = ({ page, actions }, themeOptions) => {
...
-        //if there's data in the third part, just keep the first and last part, removing the language
-        theFilePath = `${thePath}.${ext}`;
-      } else {
-        //if there's no content in the third part, it means that there's no language part. No need to remove the language
+      let fileName = path.basename(mdxFile);
+
+      let ext = path.extname(mdxFile);
+
+      if (ext !== '.mdx') {
+        console.error(`Unexpected file format in mdx path parsing: ${mdxFile}`);
```
Comment (cesarvarela) [inline site/gatsby-site/plugins/gatsby-theme-i18n/gatsby-node.js:233]: Should we crash the build instead? Logs like this probably get ignored.

(or should be logged as warnings)

File: site/gatsby-site/plugins/gatsby-theme-i18n/gatsby-node.js [hunk 2]
```diff
@@ -220,27 +220,45 @@ exports.onCreatePage = ({ page, actions }, themeOptions) => {
...
+
+      if (fileNamePieces.length == 2) {
+        // ex: index.mdx
         theFilePath = mdxFile;
+      } else if (fileNamePieces.length == 3) {
+        // ex: index.es.mdx
+        // Keep everything except the language code.
+        theFilePath = `${path.join(thePath, fileNamePieces.at(0))}${ext}`;
+      } else {
+        console.error(`Unexpected file format in mdx path parsing: ${mdxFile}`);
```
Comment (cesarvarela) [inline site/gatsby-site/plugins/gatsby-theme-i18n/gatsby-node.js:247]: ditto

File: site/gatsby-site/plugins/gatsby-theme-i18n/gatsby-node.js (hunk 1 above)
Comment (kepae) [reply site/gatsby-site/plugins/gatsby-theme-i18n/gatsby-node.js:233]: Hard agree. What's the best way to crash here?

Comment (kepae) [reply site/gatsby-site/plugins/gatsby-theme-i18n/gatsby-node.js]: Chose `throw` since there's not a specific gatsby hook (like I was expecting).

**Pull request merged**